astropy/utils/iers/iers.py

```diff
@@ -798,5 +798,7 @@ def _substitute_iers_b(cls, table):
             table['UT1_UTC_B'][:n_iers_b] = iers_b['UT1_UTC']
             table['PM_X_B'][:n_iers_b] = iers_b['PM_x']
             table['PM_Y_B'][:n_iers_b] = iers_b['PM_y']
+            table['dX_2000A_B'][:n_iers_b] = iers_b['dX_2000A']
+            table['dY_2000A_B'][:n_iers_b] = iers_b['dY_2000A']
```

Comment:
Is unit conversion handled correctly here? I suppose the test verifies that the unit matches the number, but is it asking for trouble to have these values in a different unit in exceptional circumstances? How often do users do .value without a .to(units) first because they "know" the units the quantity is expressed in? I know the PINT source code contains a fair number of .value without .to()... would just adding a .to(u.marcsec) here reduce user error? (For some reason the A and B tables express these quantities in different units, correctly described in their READMEs.)

Reply:
These are `QTable`, so their columns are just `Quantity`, and they will do the unit conversion correctly. The reason that previously explicit `.to(unit).value` were there was that we had this dance between `Table`, whiose `Column` can do masked values well but not units, and `QTable`, for which it is vice versa.

In any case, the units of the actual `table` columns here will remain the same as they always were (those of `IERS_A`), so in that sense there is no risk.  If we go the route you suggested of mimicking an IERS_A even when the input is a pure IERS_B, then we may want to reconsider.